cmd/metrics-prometheus-collector/cmd/cmd.go

```diff
@@ -313,14 +284,67 @@ func (ms *MetricsServer) Start(ctx context.Context) error {
 		log.V(1).Info("starting metrics-prometheus-collector", "MetricsPrometheusCollector", val)
 	}
 
-	// don't pre-compute and cache the metrics until we are the leader otherwise they may be ancient
-	// when we are elected
+	ctrlmetrics.Registry.MustRegister(ms.Col)
+
+	// start pre-computing metrics eagerly.  we won't publish them until we are the leader.
 	go ms.Col.Run(ctx)
 
-	ctrlmetrics.Registry.MustRegister(ms.Col)
+	if options.LeaderElection {
+		ms.doLeaderElection(ctx)
+	} else {
+		electedMetric.WithLabelValues(os.Getenv("POD_NAME")).Set(1)
+		ms.Col.IsLeaderElected.Store(true)
```

Comment:
This line seems redundant, since we already set this to `!options.LeaderElection`.